WordPress/Classes/Services/BlogService+Deduplicate.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+
+extension BlogService {
+    /// Removes any duplicate blogs in the given account
+    ///
+    /// We consider a blog to be a duplicate of another if they have the same dotComID.
+    /// For each group of duplicate blogs, this will delete all but one, giving preference to
+    /// blogs that have local drafts.
+    ///
+    /// If there's more than one blog in each group with local drafts, those will be reassigned
+    /// to the remaining blog.
+    ///
+    @objc(deduplicateBlogsForAccount:)
+    func deduplicateBlogs(for account: WPAccount) {
+        // Group all the account blogs by ID so it's easier to find duplicates
+        let blogsById = Dictionary(grouping: account.blogs, by: { $0.dotComID?.intValue ?? 0 })
+        // For any group with more than one blog, remove duplicates
+        for (blogID, group) in blogsById where group.count > 1 {
+            assert(blogID > 0, "There should not be a Blog without ID if it has an account")
```

Comment:
Wouldn't this crash in cases where a XML-RPC blog would get duplicated? They will have a `dotComID` of nil, which would get coalesced to `0` in the line above and then trigger this `assert`.

Reply:
Ah, but this is based on `Account` anyway, so they have to be `REST` blogs anyway. Nevermind.